src/infrastructure/ui/shared/component/item/meal-products-item/meal-products-item.tsx

```diff
@@ -74,25 +99,92 @@ const MealProductsItem = ({ id, name, brand, consumedQuantity, score, image, sty
                         </View>
                         <GenericButton
                             title="Modifier la quantité consommée"
-                            onPress={() => onPressEditMealProductQuantity()}
+                            onPress={onPressEditMealProductQuantity}
                             style={{
                                 container: MealProductsItemStyle.editButtonContainer,
                                 text: MealProductsItemStyle.buttonText
                             }}
                         />
                         <GenericButton
                             title="Supprimer ce produit consommé"
-                            onPress={() => onPressDeleteMealProduct(id)}
+                            onPress={onPressDeleteModal}
                             style={{
                                 container: MealProductsItemStyle.deleteButtonContainer,
                                 text: MealProductsItemStyle.buttonText
                             }}
                         />
                     </Animated.View>
                 )}
+
+                {isEditModalOpen && (
+                    <CustomModal
+                        isVisible={isEditModalOpen}
+                        dispatch={setIsEditModalOpen}
+                        title={'Modifier la quantité\n consommée'}
+                        titleSize={22}>
+                        <GenericInputWithSearchIconAndEndText
+                            placeHolder={iswater ? '25' : '100'}
+                            endText={iswater ? 'cl' : 'g'}
+                            style={MealProductsItemStyle.customModalChildren}
+                            input={consumedQuantity}
+                            dispatch={setQuantity}
+                            onPressSearchIcon={onPressValidateEditModal}
+                        />
+
+                        {servingQuantity ? (
+                            <GenericButton
+                                title={'Ajouter une portion'}
+                                onPress={onPressAddServing}
+                                style={{
+                                    container: MealProductsItemStyle.quantityModalButtonContainer,
+                                    text: MealProductsItemStyle.quantityModalButtonText
+                                }}
+                            />
+                        ) : (
+                            <></>
+                        )}
```

Comment:
```suggestion
                        {servingQuantity && (
                            <GenericButton
                                title={'Ajouter une portion'}
                                onPress={onPressAddServing}
                                style={{
                                    container: MealProductsItemStyle.quantityModalButtonContainer,
                                    text: MealProductsItemStyle.quantityModalButtonText
                                }}
                            />
                        )}
```

Le retours null ne sert à rien ici